### PR TITLE
abtj using pj.c ##refactor

### PR DESCRIPTION
**Pull request by a-meth**

### CODE REVIEW

File: libr/core/cmd_anal.c [hunk 1]
```diff
@@ -7432,39 +7433,54 @@ static void cmd_anal_abt(RCore *core, const char *input) {
...
 				r_list_foreach (path, bbi, bb) {
-					if (json && !first_bb) {
-						r_cons_printf (", ");
-					}
 					r_cons_printf ("0x%08" PFMT64x, bb->addr);
-					if (!json) {
-						r_cons_printf ("\n");
-					}
-					first_bb = false;
+					r_cons_println ("");
```
Comment (radare) [inline libr/core/cmd_anal.c]: why

File: libr/core/cmd_anal.c [hunk 2]
```diff
@@ -7432,39 +7433,54 @@ static void cmd_anal_abt(RCore *core, const char *input) {
...
-			}
+			pj_a (pj);
 			r_list_foreach (paths, pathi, path) {
-				if (json) {
-					if (!first_path) {
-						r_cons_printf (", ");
-					}
+				pj_a (pj);
+				r_list_foreach (path, bbi, bb) {
+					snprintf (bb_addr, sizeof (bb_addr), "0x%08" PFMT64x, bb->addr);
```
Comment (radare) [inline libr/core/cmd_anal.c]: why? use pj_n (bb-addr);

Comment (a-meth) [reply libr/core/cmd_anal.c]: But will that print 8 digits in hex with a 0x in the front?

File: libr/core/cmd_anal.c (hunk 1 above)
Comment (a-meth) [reply libr/core/cmd_anal.c]: Was there in the original code. What should I use to separate between two addresses?

**Comment by a-meth** [comment]
Done.

File: libr/core/cmd_anal.c [hunk 3]
```diff
@@ -7435,36 +7436,49 @@ static void cmd_anal_abt(RCore *core, const char *input) {
...
+			RAnalBlock *bb;
+			RList *path;
+			RListIter *pathi;
+			RListIter *bbi;
+			r_list_foreach (paths, pathi, path) {
 				r_list_foreach (path, bbi, bb) {
-					if (json && !first_bb) {
-						r_cons_printf (", ");
-					}
 					r_cons_printf ("0x%08" PFMT64x, bb->addr);
```
Comment (radare) [inline libr/core/cmd_anal.c]: i dont get why you do r_cons_printf and then r_cons_newline, wen you can just add \n at the end of this printf

**Pull request merged**